client/my-sites/plans-features-main/hooks/use-free-hosting-trial-assignment.ts

```diff
@@ -1,23 +1,31 @@
 import { isEnabled } from '@automattic/calypso-config';
+import { useSelect } from '@wordpress/data';
+import { ONBOARD_STORE } from 'calypso/landing/stepper/stores';
 import { useExperiment } from 'calypso/lib/explat';
+import type { OnboardSelect } from '@automattic/data-stores';
 import type { PlansIntent } from 'calypso/my-sites/plans-grid/hooks/npm-ready/data-store/use-grid-plans';
 
 export function useFreeHostingTrialAssignment( intent: PlansIntent | undefined ): {
 	isLoadingHostingTrialExperiment: boolean;
 	isAssignedToHostingTrialExperiment: boolean;
 } {
+	const hasHostingTrialOnboardingFlag = useSelect(
+		( select ) => ( select( ONBOARD_STORE ) as OnboardSelect ).isHostingTrialAvailable(),
+		[]
+	);
+
+	const isForcedToTreatment = isEnabled( 'plans/hosting-trial' ) || hasHostingTrialOnboardingFlag;
+
 	const [ isLoadingHostingTrialExperiment, experimentAssignment ] = useExperiment(
 		'wpcom_hosting_business_plan_free_trial_v2',
 		{
-			isEligible: intent === 'plans-new-hosted-site' && ! isEnabled( 'plans/hosting-trial' ),
+			isEligible: intent === 'plans-new-hosted-site' && ! isForcedToTreatment,
```

Comment:
That's a good observation. We're reading the flag into the onboarding store in the flow's `useSideEffect` hook.
https://github.com/Automattic/wp-calypso/blob/b54070ebda192e755ab3b599e763cd1b46e6946c/client/landing/stepper/declarative-flow/new-hosted-site-flow.ts#L118

In practice this is being called before the plans grid gets rendered and so we're fine. But it's possible there could be internal changes to [`FlowRenderer`](https://github.com/Automattic/wp-calypso/blob/b54070ebda192e755ab3b599e763cd1b46e6946c/client/landing/stepper/declarative-flow/internals/index.tsx#L46) in the future that would change that. But I think if `useSideEffect` started to run after components were rendered in the future then that'd break a lot of stepper flows, not just us.

So I think it's ok to assume this will keep working in the future. If we were super paranoid we could have the hook return `isLoadingHostingTrialExperiment: false` until we knew the flag in the onboarding store had been initialised.